subprojects/plugins/src/main/java/org/gradle/api/internal/plugins/StartScriptTemplateBindingFactory.java

```diff
@@ -59,8 +59,8 @@ public Map<String, String> transform(JavaAppStartScriptGenerationDetails details
         binding.put(ScriptBindingParameter.APP_HOME_REL_PATH.getKey(), createJoinedAppHomeRelativePath(details.getScriptRelPath()));
         binding.put(ScriptBindingParameter.CLASSPATH.getKey(), createJoinedPath(details.getClasspath()));
         binding.put(ScriptBindingParameter.MODULE_PATH.getKey(), createJoinedPath(details.getModulePath()));
+        binding.put(ScriptBindingParameter.CURRENT_YEAR.getKey(), details.getCurrentYear());
```

Comment:
❓ This might go against reproducible builds. The wrapper scripts for a given Gradle version will change every year. Same goes for applications. We should strive for reproducibility by default. Could this be considered as part of this change?

Reply:
I wasn't considering the wrapper scripts to be artifacts that needed to be reproducible.  Are many people actually re-generating these with each build?  Why?

Reply:
These scripts are regenerated when you run the `:wrapper` task during an upgrade.

We update the content of these scripts sometime and it is part of a regular upgrade.

We don't provide checksums for these scripts but we should to simplify reviews for changes in these scripts in builds that don't customize them.